crates/pallets/starknet/src/lib.rs

```diff
@@ -721,24 +725,29 @@ pub mod pallet {
             let sequencer_address = ContractAddressWrapper::default();
             let block_timestamp = Self::block_timestamp();
             let transaction_count = pending.len() as u128;
+            let transactions: Vec<Transaction> = pending.into_iter().map(|(transaction, _)| transaction).collect();
+            let transactions_slice: &[Transaction] = &transactions;
             let (transaction_commitment, (event_commitment, event_count)) =
-                commitment::calculate_commitments::<PedersenHasher>(&pending);
+                commitment::calculate_commitments::<PedersenHasher>(transactions_slice);
             let protocol_version = None;
             let extra_data = None;
 
-            let block = StarknetBlock::new(StarknetHeader::new(
-                parent_block_hash,
-                block_number,
-                global_state_root,
-                sequencer_address,
-                block_timestamp,
-                transaction_count,
-                transaction_commitment,
-                event_count,
-                event_commitment,
-                protocol_version,
-                extra_data,
-            ));
+            let block = StarknetBlock::new(
```

Comment:
add new test / update existing tests to cover the change of this PR, i.e check presence of transactions as well as receipt, etc